packages/host/app/components/matrix/room-message.gts

```diff
@@ -151,71 +166,32 @@ export default class RoomMessage extends Component<Signature> {
   @service private declare operatorModeStateService: OperatorModeStateService;
   @service private declare matrixService: MatrixService;
   @service declare commandService: CommandService;
-  @tracked loadedResources:
-    | {
-        cards: CardDef[] | undefined;
-        files: FileDef[] | undefined;
-        errors: { id: string; error: Error }[] | undefined;
-      }
-    | undefined;
-
-  loadMessageResources = task(async () => {
-    let cards: CardDef[] = [];
-    let errors: { id: string; error: Error }[] = [];
-
-    let promises = this.message
-      .attachedResources(this)
-      ?.map(async (resource) => {
-        await resource.loaded;
-        if (resource.card) {
-          cards.push(resource.card);
-        } else if (resource.cardError) {
-          let {
-            id,
-            message,
-            title: name,
-            meta: { stack },
-          } = resource.cardError;
-          if (id) {
-            errors.push({
-              id,
-              error: { name, message, stack: stack ?? undefined },
-            });
-          }
-        }
-      });
-
-    if (promises) {
-      await Promise.all(promises);
+
+  @cached
+  private get areResourcesLoaded() {
+    if (!this.attachedCardResources) {
+      return false;
     }
 
-    this.loadedResources = {
-      cards: cards.length ? cards : undefined,
-      files: this.message.attachedFiles?.length
-        ? this.message.attachedFiles
-        : undefined,
-      errors: errors.length ? errors : undefined,
-    };
-  });
+    return this.attachedCardResources.length === 0
+      ? true
+      : this.attachedCardResources.every((r) => r.isLoaded);
+  }
 
+  @cached
   private get errorMessage() {
     if (this.message.errorMessage) {
       return this.message.errorMessage;
     }
     if (this.streamingTimeout) {
       return 'This message was processing for too long. Please try again.';
     }
-    if (!this.loadedResources?.errors) {
+    let resourcesWithErrors = (this.attachedCardResources ?? []).filter(
+      (r) => r.cardError,
+    );
+    if (resourcesWithErrors.length === 0) {
       return undefined;
     }
-
-    let hasResourceErrors = this.loadedResources.errors.length > 0;
-    if (hasResourceErrors) {
-      return 'Error rendering attached cards.';
-    }
-
-    return this.loadedResources.errors
-      .map((e: { id: string; error: Error }) => `${e.id}: ${e.error.message}`)
-      .join(', ');
```

Comment:
it was impossible to get to this place in the code, as we would always return above in this scenario, so i opted to remove it.